From d3621630415ea3f2646238520b6ffb89aebe16df Mon Sep 17 00:00:00 2001
From: "eldin.lee" <eldin.lee@quantatw.com>
Date: Wed, 9 Jul 2025 14:42:07 +0800
Subject: [PATCH] Add delay before leak event log to prevent AC OFF glitches

Insert a short delay before logging to prevent transient AC power loss from triggering incorrect events.
---
 src/leakdetector/LeakEvents.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/leakdetector/LeakEvents.cpp b/src/leakdetector/LeakEvents.cpp
index 290e972..4539f4d 100644
--- a/src/leakdetector/LeakEvents.cpp
+++ b/src/leakdetector/LeakEvents.cpp
@@ -22,6 +22,8 @@ auto Events::generateLeakEvent(sdbusplus::object_path detectorPath,
 {
     auto eventName = std::make_tuple(detectorPath.str, level);
 
+    co_await sdbusplus::async::sleep_for(ctx, std::chrono::milliseconds(500));
+
     // NOLINTNEXTLINE(clang-analyzer-core.uninitialized.Branch)
     if (state == DetectorStateIntf::DetectorState::Normal)
     {
-- 
2.53.0-Meta

